packages/toolpad-app/src/components/EditableText.tsx

```diff
@@ -1,39 +1,94 @@
 import * as React from 'react';
-import { Skeleton, TextField, Typography, TypographyVariant, Tooltip } from '@mui/material';
+import {
+  TextField,
+  Theme,
+  TypographyVariant,
+  SxProps,
+  InputBaseComponentProps,
+} from '@mui/material';
 
 interface EditableTextProps {
   defaultValue?: string;
-  loading: boolean;
-  editing: boolean;
-  isError: boolean;
+  editing?: boolean;
+  isError?: boolean;
   errorText?: string;
-  onKeyUp: (event: React.KeyboardEvent<HTMLInputElement>) => void;
-  onBlur: (event: React.FocusEvent<HTMLInputElement>) => void;
+  onKeyUp?: (event: React.KeyboardEvent<HTMLInputElement>) => void;
```

Comment:
I'm not super fond of leaking these key handlers. Would it be make sense to build this using controlled properties only? i.e. `value` and `onChange` that act like a regular `TextField`. And to change editable state, `open` and `onClose`, analog to e.g. the MUI `Dialog`, `Menu`, `Drawer`,...